### PR TITLE
improve sanity of securitymanager file permissions

### PR DESCRIPTION
**Pull request by rmuir**
conf/ and plugins/ do not need read-write access: this prevents a lot
of bad possibilities.

### CODE REVIEW

**Comment by jaymode** [comment]
LGTM

File: core/src/test/java/org/elasticsearch/bootstrap/SecurityTests.java [hunk 1]
```diff
@@ -60,6 +60,9 @@ public void testGeneratedPermissions() throws Exception {
         assertFalse(permissions.implies(new FilePermission(path.resolve("other").toString(), "read")));
         // double check we overwrote java.io.tmpdir correctly for the test
         assertFalse(permissions.implies(new FilePermission(realTmpDir.toString(), "read")));
+        // a read only directory
+        assertTrue(permissions.implies(new FilePermission(esHome.resolve("config").toString(), "read")));
+        assertFalse(permissions.implies(new FilePermission(esHome.resolve("config").toString(), "write")));
```
Comment (kimchy) [inline core/src/test/java/org/elasticsearch/bootstrap/SecurityTests.java]: should we do the same check for plugins dir as well?

**Comment by kimchy** [comment]
LGTM, added a comment around testing for plugins as well

**Comment by rmuir** [comment]
I pushed a commit improving this test (it did really need it)

**Comment by rjernst** [comment]
Nice! LGTM too

**Pull request merged**